com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/persistence/service/ConfigurationEntryService.java

```diff
@@ -0,0 +1,81 @@
+import com.sap.cloud.lm.sl.cf.core.persistence.object.factory.PersistenceObjectFactory;
+import com.sap.cloud.lm.sl.cf.core.persistence.query.ConfigurationEntryQuery;
+import com.sap.cloud.lm.sl.cf.core.persistence.query.impl.ConfigurationEntryQueryImpl;
+import com.sap.cloud.lm.sl.common.ConflictException;
+import com.sap.cloud.lm.sl.common.NotFoundException;
+
+@Named
+public class ConfigurationEntryService extends PersistenceService<ConfigurationEntry, ConfigurationEntryDto, Long> {
+
+    @Inject
+    protected ConfigurationEntryFactory entryFactory;
+
+    @Inject
+    public ConfigurationEntryService(EntityManagerFactory entityManagerFactory) {
+        super(entityManagerFactory);
+    }
+
+    public ConfigurationEntryQuery createQuery() {
+        return new ConfigurationEntryQueryImpl(createEntityManager(), entryFactory);
+    }
+
+    @Override
+    protected ConfigurationEntryDto merge(ConfigurationEntryDto existingEntry, ConfigurationEntryDto newEntry) {
+        super.merge(existingEntry, newEntry);
+        String providerNid = ObjectUtils.firstNonNull(removeDefault(newEntry.getProviderNid()), existingEntry.getProviderNid());
+        String providerId = ObjectUtils.firstNonNull(newEntry.getProviderId(), existingEntry.getProviderId());
+        String targetOrg = ObjectUtils.firstNonNull(newEntry.getTargetOrg(), existingEntry.getTargetOrg());
+        String targetSpace = ObjectUtils.firstNonNull(newEntry.getTargetSpace(), existingEntry.getTargetSpace());
+        String providerVersion = ObjectUtils.firstNonNull(removeDefault(newEntry.getProviderVersion()), existingEntry.getProviderVersion());
+        String content = ObjectUtils.firstNonNull(newEntry.getContent(), existingEntry.getContent());
+        String visibility = ObjectUtils.firstNonNull(newEntry.getVisibility(), existingEntry.getVisibility());
+        String spaceId = ObjectUtils.firstNonNull(newEntry.getSpaceId(), existingEntry.getSpaceId());
+        return new ConfigurationEntryDto(newEntry.getPrimaryKey(),
+                                         providerNid,
+                                         providerId,
+                                         providerVersion,
+                                         targetOrg,
+                                         targetSpace,
+                                         content,
+                                         visibility,
+                                         spaceId);
+    }
+
+    private String removeDefault(String value) {
+        return value.equals(PersistenceMetadata.NOT_AVAILABLE) ? null : value;
+    }
+
+    @Override
+    protected PersistenceObjectFactory<ConfigurationEntry, ConfigurationEntryDto> getPersistenceObjectFactory() {
+        return entryFactory;
+    }
+
+    @Override
+    protected void onEntityConflict(ConfigurationEntryDto entry, Throwable t) {
+        throw (ConflictException) new ConflictException(Messages.CONFIGURATION_ENTRY_ALREADY_EXISTS,
```

Comment:
Because of 'initCause' which returns Throwable. However this will be removed as there's an appropriate constructor for this, removing the initCause call.